AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/MAX/Native Ads/ALMAXTemplateNativeAdViewController.m

```diff
@@ -0,0 +1,110 @@
+//
+//  ALMAXTemplateNativeAdViewController.m
+//  AppLovin MAX Demo App - ObjC
+//
+//  Created by Billy Hu on 1/20/22.
+//  Copyright © 2022 AppLovin Corporation. All rights reserved.
+//
+
+#import "ALMAXTemplateNativeAdViewController.h"
+#import "ALBaseAdViewController.h"
+#import <Adjust/Adjust.h>
+#import <AppLovinSDK/AppLovinSDK.h>
+
+@interface ALMAXTemplateNativeAdViewController()<MANativeAdDelegate, MAAdRevenueDelegate>
+
+@property (nonatomic, weak) IBOutlet UIView *nativeAdContainerView;
+
+@property (nonatomic, strong) MANativeAdLoader *nativeAdLoader;
+@property (nonatomic, strong) UIView *nativeAdView;
+@property (nonatomic, strong, nullable) MAAd *nativeAd;
+
+@end
+
+@implementation ALMAXTemplateNativeAdViewController
+
+#pragma mark - View Lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.nativeAdLoader = [[MANativeAdLoader alloc] initWithAdUnitIdentifier: @"YOUR_AD_UNIT"];
+    self.nativeAdLoader.nativeAdDelegate = self;
+    self.nativeAdLoader.revenueDelegate = self;
+}
+
+- (void)dealloc
+{
+    [self cleanUpAdIfNeeded];
+    
+    self.nativeAdLoader.nativeAdDelegate = nil;
+    self.nativeAdLoader.revenueDelegate = nil;
+}
+
+- (void)cleanUpAdIfNeeded
+{
+    // Clean up any pre-existing native ad to prevent memory leaks
+    if ( self.nativeAd )
+    {
+        [self.nativeAdLoader destroyAd: self.nativeAd];
+    }
+
+    if ( self.nativeAdView )
+    {
+        [self.nativeAdView removeFromSuperview];
+    }
+}
+
+#pragma mark - IB Actions
+
+- (IBAction)showAd
+{
+    [self cleanUpAdIfNeeded];
+    
+    [self.nativeAdLoader loadAd];
+}
+
+#pragma mark - NativeAdDelegate Protocol
+
+- (void)didLoadNativeAd:(MANativeAdView *)nativeAdView forAd:(MAAd *)ad
+{
+    [self logCallback: __PRETTY_FUNCTION__];
+    
+    // Save ad for cleanup
+    self.nativeAd = ad;
+    
+    // Add ad view to view
+    self.nativeAdView = nativeAdView;
+    [self.nativeAdContainerView addSubview: nativeAdView];
```

Comment:
I'm pretty sure you need to call `al_pinToSuperview` on `nativeAdView` to pin it to the superview so it re-sizes accordingly on rotation, etc.

Reply:
`al_pinToSuperview` is a `UIView` category method we exposed publicly. See `ALUtils.h` in our SDK's public API.